packages/cli/src/generate/generators/app/create-app.ts

```diff
@@ -46,6 +86,14 @@ export async function createApp({ name, sessionSecret, autoInstall }:
     sessionSecret: sessionSecret ? sessionSecret : crypto.randomBytes(16).toString('hex')
   };
 
+  // Validating whether if the project-name follows npm naming conventions
+  if (!validateProjectName(name)) {
+    console.log(
+      red(`\n ${red(`${name} doesn't follow the npm naming conventions. Kindly give a vaild project-name`)}`)
+    );
+    process.exit(1);
```

Comment:
Using `process.exit` just after a `console.log` may make the message not to be printed in some situations (see https://github.com/FoalTS/foal/pull/309#discussion_r247313980). Could we use a `return` instead? Also, using `process.exit` in the code may make the test process to exit unexpectedly when running the unit tests.